transport/socks5/socks5.go

```diff
@@ -119,3 +149,45 @@ func appendSOCKS5Address(b []byte, address string) ([]byte, error) {
 	b = binary.BigEndian.AppendUint16(b, uint16(portNum))
 	return b, nil
 }
+
+func readAddr(r io.Reader) (*address, error) {
+	address := &address{}
+
+	var addrType [1]byte
+	if _, err := r.Read(addrType[:]); err != nil {
+		return nil, err
+	}
+
+	switch addrType[0] {
+	case addrTypeIPv4:
+		var addr [4]byte
+		if _, err := io.ReadFull(r, addr[:]); err != nil {
+			return nil, err
+		}
+		address.IP = netip.AddrFrom4(addr)
+	case addrTypeIPv6:
+		var addr [16]byte
+		if _, err := io.ReadFull(r, addr[:]); err != nil {
+			return nil, err
+		}
+		address.IP = netip.AddrFrom16(addr)
+	case addrTypeDomainName:
+		if _, err := r.Read(addrType[:]); err != nil {
+			return nil, err
+		}
+		addrLen := int(addrType[0])
+		fqdn := make([]byte, addrLen)
```

Comment:
Make sure addrLen is of type byte, to limit the allocation.
```suggestion
		addrLen := addrType[0]
		fqdn := make([]byte, int(addrLen))
```

Reply:
@fortuna we need to make sure `addrLen` is under 255 to prevent large memory allocation. The following code prevents this and also throws a meaningful error message is length is more than 255
 
```go
		addrLen := int(addrType[0])
		if addrLen > 255 {
			return nil, fmt.Errorf("domain name length = %v is over 255", addrLen)
		}
		fqdn := make([]byte, addrLen)
		if _, err := io.ReadFull(r, fqdn); err != nil {
			return nil, err
		}
```

Reply:
This is not needed if you keep the type of addrLen as byte.
Why are you converting to int?